src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/BitVector256.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    internal unsafe struct BitVector256
+    {
+        private fixed uint _values[8];
+
+        public void Set(int c)
+        {
+            Debug.Assert(c < 256);
+            uint offset = (uint)(c >> 5);
+            uint significantBit = 1u << (c & 31);
```

Comment:
I think technically the `& 31` isn't necessary, but the JIT should also remove the duplication such that it won't negatively affect throughput.

Reply:
Looks like the JIT is able to remove it here. I'll update the code for good measure.